src/utils/hooks/useDebounce.ts

```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect } from 'react';
+
+export default function useDebounce(value: string, delay : number) {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+
+  return debouncedValue;
+}
```

Comment:
오.. 직접 만든건가? 성장했구만.. 이렇게 좀 더 타입을 잘 사용할 수 있을 듯 합니다. delay 는 대체적으로 500을 많이 써서 기본 값을 줘봤는데, 기존대로 안줘도 갠춘할 것 같슴니다.

```suggestion
function useDebounce<T>(value: T, delay?: number): T {
  const [debouncedValue, setDebouncedValue] = useState<T>(value)

  useEffect(() => {
    const handler = setTimeout(() => setDebouncedValue(value), delay || 500)

    return () => {
      clearTimeout(handler);
    }
  }, [value, delay]);

  return debouncedValue;
}
```

Reply:
오..제네릭..이번 자바에서 배운게..여기에서도 사용가능하군요..? 수정하겠습니다!